File: ZendAMF/app/servicos/ServicosFormadePagto.php
<?php

include 'app/vo/FormadePagtoVO.php';

class ServicosFormadePagto {
	
	
	private $conn;
	
	function ServicosFormadePagto() {
		$db = new BaseDados();
		$this->conn = $db->conn;
	}
	
	public function addFormadePagto(FormadePagtoVO $formadepagto){
	 	
		$sql = "insert into formadepgto (descricao) values ('{$formadepagto->descricao}')";
		
		$resultado = $this->conn->Execute($sql);
		$formadepagto->codigo = $this->conn->insert_Id();
		return $formadepagto;
		
	}
	
	public function removerFormadePagto(FormadePagtoVO $formadepagto){
		
		$sql = "delete from formadepgto  where codigo = {$formadepagto->codigo}";
		$resultado = $this->conn->Execute($sql);
		if($this->conn->Affected_Rows()==0){
			return false;
		}
		return true;
	}
	
	public function pesquisarFormadePagto($texto,$coluna){
		$sql = "select * from formadepgto where $coluna like '%$texto%'";
		$resultado = $this->conn->Execute($sql);
		while($registro = $resultado->FetchNextObject()){			
			$dados_formadepagto = new FormadePagtoVO();
			$dados_formadepagto->codigo = $registro->CODIGO;
			$dados_formadepagto->descricao = $registro->DESCRICAO;
			$retorna_dados_formadepagto [] = $dados_formadepagto;
		}
		return $retorna_dados_formadepagto;
	}
	
	public function getFormadePagtos(){
		$sql = "select * from formadepgto";
		
		$resultado = $this->conn->Execute($sql);
		
		while($registro = $resultado->FetchNextObject()){			
			$dados_formadepagto = new FormadePagtoVO();
			$dados_formadepagto->codigo = $registro->CODIGO;
			$dados_formadepagto->descricao = $registro->DESCRICAO;
			$retorna_dados_formadepagto [] = $dados_formadepagto;
		}
		return $retorna_dados_formadepagto;	
	}
}


//$eu = new ServicosFormadePagto();
//$formadepagto = new FormadePagtoVO();
//$formadepagto->codigo = 2;
//echo $eu->removerFormadePagto($formadepagto);

?>

File: trunk/ServidorJava/WEB-INF/src/services/ServicosFormadePagto.java
package services;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import vo.FormadePagtoVO;

public class ServicosFormadePagto {

	public FormadePagtoVO addFormadePagto(FormadePagtoVO formadepgto) {

		String sql = "insert into formadepgto (descricao) values ('"
				+ formadepgto.descricao + "')";
		Statement st;
		this.banco.conectar();
		this.banco.getConexao().setAutoCommit(false);
		st = this.banco.getConexao().createStatement();
		if (st.executeUpdate(sql, Statement.RETURN_GENERATED_KEYS) == 0) {
			throw new RuntimeException("Erro ao salvar Forma de pagamento!");
		}
		ResultSet rs = st.getGeneratedKeys();
		if (rs.next())
			formadepgto.codigo = rs.getInt(1);
		else
			throw new RuntimeException("Erro ao salvar Forma de pagamento!");
		this.banco.getConexao().commit();
		return formadepgto;

	}

	public boolean removerFormadePagto(FormadePagtoVO formadepgto) {
		String sql = "delete from formadepgto  where codigo = "
				+ formadepgto.codigo + "";
		if (banco.executarNoQuery(sql) == 0) {
			return false;
		}
		return true;
	}

	public FormadePagtoVO atualizarFormadePagto(FormadePagtoVO formaPagamento) {
		String sql = "UPDATE formadepgto SET descricao = '"
				+ formaPagamento.descricao + "' where codigo = "
				+ formaPagamento.codigo + "";
		if (banco.executarNoQuery(sql) == 0) {
			throw new RuntimeException("Erro ao atualizar forma de pagamento!");
		}
		return formaPagamento;

	}

	public ArrayList<FormadePagtoVO> pesquisarFormadePagto(String texto,
			String coluna) {
		String sql = "select * from formadepgto where " + coluna + " like '%"
				+ texto + "%'";
		ResultSet rs = banco.executar(sql);
		return toFormaPagto(rs);
	}

	public ArrayList<FormadePagtoVO> getFormadePagtos() {
		String sql = "select * from formadepgto";
		ResultSet rs = banco.executar(sql);
		return toFormaPagto(rs);
	}

	private ArrayList<FormadePagtoVO> toFormaPagto(ResultSet rs)
			throws SQLException {
		ArrayList<FormadePagtoVO> fp = new ArrayList<FormadePagtoVO>();
		while (rs.next()) {
			FormadePagtoVO dados_item = new FormadePagtoVO();
			dados_item.codigo = rs.getInt("codigo");
			dados_item.descricao = rs.getString("descricao");
			fp.add(dados_item);
		}
		return fp;

	}

	public boolean isExiste(FormadePagtoVO formadepgto) {
		String sql = "select * from formadepgto where codigo = '"
				+ formadepgto.codigo + "'";
		ResultSet rs;
		rs = this.banco.executar(sql);
		if (rs.next())
			return true;
		else
			return false;

	}

}
